hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheOperation.java

```diff
@@ -139,4 +137,25 @@ public final int getSyncBackupCount() {
     public final int getAsyncBackupCount() {
         return cache != null ? cache.getConfig().getAsyncBackupCount() : 0;
     }
+
+    protected final void publishWanUpdate(Data dataKey) {
+        if (!cache.isWanReplicationEnabled()) {
+            return;
+        }
+
+        CacheRecord record = cache.getRecord(dataKey);
+        if (record == null) {
+            return;
+        }
+
+        NodeEngine nodeEngine = getNodeEngine();
+        SerializationService serializationService = nodeEngine.getSerializationService();
+        Data dataValue = toHeapData(serializationService.toData(record.getValue()));
+        CacheEntryView<Data, Data> entryView = createDefaultEntryView(dataKey, dataValue, record);
+        wanEventPublisher.publishWanUpdate(name, entryView);
+    }
+
+    protected final void publishWanRemove(Data dataKey) {
+        wanEventPublisher.publishWanRemove(name, dataKey);
```

Comment:
I think similarly to the update counterpart, we should check `cache.isWanReplicationEnabled` here as well and call remove only if the check returns true. It is implemented that way in `MapOperation`.

Reply:
done.